### ISSUE TITLE
`kubeadm upgrade plan` fails on restricted networks

### ISSUE THREAD

**Issue by praseodym** [kubernetes/kubeadm#498]
BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`): v1.8.1

**Environment**:
- **Kubernetes version** (use `kubectl version`): v1.7.8
- **Cloud provider or hardware configuration**: VMware
- **OS** (e.g. from /etc/os-release): Debian GNU/Linux 9 (stretch)
- **Kernel** (e.g. `uname -a`): Linux 4.9.0-4-amd64 #1 SMP Debian 4.9.51-1 (2017-09-28) x86_64 GNU/Linux

## What happened?
When running `kubeadm upgrade plan` in a network where https://dl.k8s.io is not reachable (e.g. due to firewall restrictions), a `FATAL` error will occur while fetching the latest stable version.

## What you expected to happen?
The check should fail in a non-fatal manner, e.g. get logged as a warning.

## How to reproduce it (as minimally and precisely as possible)?
Make dl.k8s.io unreachable (e.g. `echo 255.0.0.0 dl.k8s.io >> /etc/hosts`) or simply go offline, then execute `kubeadm upgrade plan`:

```
# ./kubeadm upgrade plan
[preflight] Running pre-flight checks
[upgrade] Making sure the cluster is healthy:
[upgrade/health] Checking API Server health: Healthy
[upgrade/health] Checking Node health: All Nodes are healthy
[upgrade/health] Checking Static Pod manifests exists on disk: All manifests exist on disk
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -oyaml'
[upgrade] Fetching available versions to upgrade to:
[upgrade/versions] Cluster version: v1.7.8
[upgrade/versions] kubeadm version: v1.8.1
[upgrade/versions] FATAL: Couldn't fetch latest stable version version from the internet: unable to get URL "https://dl.k8s.io/release/stable.txt": Get https://dl.k8s.io/release/stable.txt: dial tcp 255.0.0.0:443: i/o timeout
```

**Comment by erhudy** [comment]
Same problem here. IMO, if the version is specified in the ConfigMap, it should skip the version check.

**Comment by luxas** [comment]
We'll have this fixed in an upcoming v1.8 release. Thanks for the PR, I'm looking!
